dev/src/index.ts

```diff
@@ -338,7 +340,29 @@ export class Firestore {
       libraryHeader.libVersion += ' fire/' + settings.firebaseVersion;
     }
 
-    this.validateAndApplySettings(Object.assign({}, settings, libraryHeader));
+    if (process.env.FIRESTORE_EMULATOR_HOST) {
+      validateHost(
+        'FIRESTORE_EMULATOR_HOST',
+        process.env.FIRESTORE_EMULATOR_HOST
+      );
+
+      const url = new URL('http://' + process.env.FIRESTORE_EMULATOR_HOST);
+      const host = url.hostname;
```

Comment:
In every other library, we're trying to consistently support the `apiEndpoint` property in the options that are passed into the client constructor.  Can we support this as well here as well as `host`?

Reply:
If `host` is not specified, `apiEndpoint` and `servicePath` are passed through and work as they do in other languages. If `host` and either `apiEndpoint` or `servicePath` is specified, we fail initialization.